testutils/m/match.go

```diff
@@ -39,6 +39,39 @@ func MatchRoomName(name string) RoomMatcher {
 	}
 }
 
+// MatchRoomAvatar builds a RoomMatcher which checks that the given room response has
+// set the room's avatar to the given value.
+func MatchRoomAvatar(wantAvatar string) RoomMatcher {
```

Comment:
I was trying to be very explicit about the different cases here, because the in-memory values and the serialised values are different, which is confusing.